src/backend/utils/storage.js

```diff
@@ -5,21 +5,52 @@ const FEED_ID = 'feed_id';
 const FEEDS = 'feeds';
 
 const POSTS = 'posts';
+const INACTIVE_FEEDS = 'inactive_feeds';
 
 module.exports = {
-  addFeed: async (name, url) => {
-    // "If the key does not exist, it is set to 0 before performing the operation"
-    // https://redis.io/commands/INCR
-    const feedId = await redis.incr(FEED_ID);
-    await redis
-      .multi()
-      // Using hmset() until hset() fully supports multiple fields:
-      // https://github.com/stipsan/ioredis-mock/issues/345
-      // https://github.com/luin/ioredis/issues/551
-      .hmset(feedId, 'name', name, 'url', url)
-      .sadd(FEEDS, feedId)
-      .exec();
-    return feedId;
+  /**
+   * Adds/Moves feed to the active FEEDS set and returns the feedId
+   * @param feed Feed Object
+   * @return feedId
+   */
+  addFeed: async feed => {
+    if (feed.feedId && (await redis.exists(INACTIVE_FEEDS)) === 1) {
+      await redis.smove(INACTIVE_FEEDS, FEEDS, feed.feedId);
+    } else {
+      // "If the key does not exist, it is set to 0 before performing the operation"
+      // https://redis.io/commands/INCR
+      const feedId = await redis.incr(FEED_ID);
+      await redis
+        .multi()
+        // Using hmset() until hset() fully supports multiple fields:
+        // https://github.com/stipsan/ioredis-mock/issues/345
+        // https://github.com/luin/ioredis/issues/551
+        .hmset(feedId, 'name', feed.name, 'url', feed.url)
+        .sadd(FEEDS, feedId)
+        .exec();
+      return feedId;
+    }
+    return feed.feedId;
+  },
+
+  /**
+   * Moves feed from active FEEDS set to INACTIVE_FEEDS set. A destination must always be initialized
+   * with sadd() before other members can be moved to the destination
+   * @param feed Feed Object
+   */
+  addInactiveFeed: async feed => {
+    if ((await redis.exists(INACTIVE_FEEDS)) === 1) {
```

Comment:
This code is identical to what you do above.  The only difference is that one uses `FEEDS` where the other uses `INACTIVE_FEEDS`.  As such, you can refactor this to use 2 arguments passed to a function, and then only do this code once, but call it from two helper functions that deal with the order of these args.

Reply:
I slowly realized this... and have something working in storage.js as with feedSet as a param for the set to send it to

```
addFeed: async (feed, feedSet){

}
```

Reply:
whoops please disregard this as I replied to this before the previous one.